### PR TITLE
Fix FindImplementationService not finding all implementations of the partial class

### PR DESCRIPTION
**Pull request by SirIntruder**
Currently, when you use "Find Implementations" on a partial class, it appears to randomly select one of the definitions instead of returning all of them (like VS does). Looks like this was unintentionally broken in order to handle partialMethod case (guess there is makes sense to take symbol.Locations.First() instead of symbol.Locations).

Should be easy to merge with @filipw fix for separate bug fix in the same service.

### CODE REVIEW

**Comment by filipw** [comment]
can you add a test illustrating what this is fixing?

**Comment by SirIntruder** [comment]
Will push test in later today

**Comment by SirIntruder** [comment]
Sorry for the delay. I added test for finding all implementations of the partial class and one test for partial methods (to make sure this functionality is kept intact).

**Comment by filipw** [comment]
I merged https://github.com/OmniSharp/omnisharp-roslyn/pull/1302, could you please resolve the conflicts - thanks! Otherwise, looks good to me

**Comment by SirIntruder** [comment]
@filipw done!

File: tests/OmniSharp.Roslyn.CSharp.Tests/FindImplementationFacts.cs [hunk 1]
```diff
@@ -152,6 +152,38 @@ public class MyClass
...
+            const string code = @"
+                public partial class SomeClass { partial void Some$$Method(); }
+                public partial class SomeClass { partial void SomeMethod() { /* this is implementation of the partial method */ }";
+
+            var implementations = await FindImplementationsAsync(code, filename);
+
+            Assert.Single(implementations);
+
+            var implementation = implementations.First();
+            Assert.Equal("SomeMethod", implementation.Name);
```
Comment (rchande) [inline tests/OmniSharp.Roslyn.CSharp.Tests/FindImplementationFacts.cs]: Assert on the location of `implementation` to prove you found the implementation copy?

Comment (filipw) [reply tests/OmniSharp.Roslyn.CSharp.Tests/FindImplementationFacts.cs]: this is a good idea.

Comment (david-driscoll) [reply tests/OmniSharp.Roslyn.CSharp.Tests/FindImplementationFacts.cs]: @SirIntruder if you could fix this feedback, this looks good to me.

**Comment by SirIntruder** [comment]
@rchande @filipw @david-driscoll Thanks a lot for the tip, I would actually regress partial method without it. I enhanced the test and changed handler code, looks like it works correctly now.

**Comment by filipw** [comment]
looks good, indeed this is the behavior we want 👍

**Comment by filipw** [comment]
thanks!

**Pull request merged**